src/app/shared/services/tag.service.ts

```diff
@@ -8,7 +8,7 @@ import { HttpService } from './http.service';
   providedIn: 'root'
 })
 export class TagsService {
```

Comment:
In my opinion we must have one tags service with 2 requests for all tags and top tags.